types/vote_test.go

```diff
@@ -336,39 +339,110 @@ func TestVoteString(t *testing.T) {
 	}
 }
 
-func TestVoteValidateBasic(t *testing.T) {
+func signAndMalleateVote(ctx context.Context, t *testing.T, pv PrivValidator, chainID string, malleateVote func(*Vote), vote *Vote) {
+	t.Helper()
+
+	v := vote.ToProto()
+	require.NoError(t, pv.SignVote(ctx, chainID, v))
+	vote.Signature = v.Signature
+	vote.ExtensionSignature = v.ExtensionSignature
+	malleateVote(vote)
```

Comment:
Sure, I've separated it out now. I originally did it like this because it seemed slightly neater to have this functionality handled on a single line, but I see your point.